### PR TITLE
feat(#164): Add if/else

### PR DESCRIPTION
**Pull request by doug-wade**

### CODE REVIEW

**Comment by doug-wade** [comment]
from my discussion with @supertyrelle --

what happens when we try to do an `else-if` block?

```
<t-if condition="${side === HEADS}>
  <div slot="true">heads for the hills</div>
  <div slot="false">
    <t-if condition="${side === TAILS}">
      <div slot="true">tails never fails</div>
      <div slot="false">it's on its side</div>
    </t-if>
</t-if>
```

can we have a "true" template nested inside itself?

File: packages/website/pages/core.md [hunk 1]
```diff
@@ -245,3 +245,58 @@ defineComponent({
     shadowMode: 'open',
 });
 ```
+
+## Components
```
Comment (doug-wade) [inline packages/website/pages/core.md:249]: we'll want to call out that all of these elements will add a wrapper element to the component definition, but not to the resultant dom.

File: packages/test-utils/src/util/base-wrapper.ts [hunk 2]
```diff
@@ -86,6 +86,10 @@ export default class BaseWrapper implements Wrapper {
         return this.element.outerHTML;
     }
 
+    shadowHtml() {
+        return global.shadowRootRegistry.get(this.element);
```
Comment (dougy-fresh) [inline packages/test-utils/src/util/base-wrapper.ts:90]: I think we should return a wrapper here, rather than a dom node.

**Comment by doug-wade** [comment]
This branch would also close #83

**Pull request merged**